src/fides/api/ops/service/privacy_request/consent_email_batch_service.py

```diff
@@ -0,0 +1,260 @@
+                "Skipping batch consent email send with status: {}",
+                ConsentEmailExitState.no_applicable_connectors.value,
+            )
+            return ConsentEmailExitState.no_applicable_connectors
+
+        batched_user_data: List[BatchedUserConsentData] = stage_resource_per_connector(
+            conn_configs
+        )
+        add_batched_user_preferences_to_emails(privacy_requests, batched_user_data)
+
+        if not any(
+            pending_email.batched_user_consent_preferences
+            for pending_email in batched_user_data
+        ):
+            requeue_privacy_requests_after_consent_email_send(privacy_requests, session)
+            logger.info(
+                "Skipping batch consent email send with status: {}",
+                ConsentEmailExitState.missing_required_data.value,
+            )
+            return ConsentEmailExitState.missing_required_data
+
+        try:
+            send_prepared_emails(session, batched_user_data, privacy_requests)
+        except MessageDispatchException as exc:
+            logger.error(
+                "Consent email send for connector failed with exception: '{}'",
+                exc,
+            )
+            return ConsentEmailExitState.email_send_failed
+
+    requeue_privacy_requests_after_consent_email_send(privacy_requests, session)
+    return ConsentEmailExitState.complete
+
+
+def requeue_privacy_requests_after_consent_email_send(
+    privacy_requests: Query, db: Session
+) -> None:
+    """After batch consent email send, requeue privacy requests from the post webhooks step
+    to wrap up processing and transition to a "complete" state.
+
+    Also cache on the privacy request itself that it is paused at the post-webhooks state,
+    in case something happens in re-queueing.
+    """
+    logger.info("Batched consent email send complete.")
+    logger.info("Queuing privacy requests from 'post_webhooks' step.")
+    for privacy_request in privacy_requests:
+        privacy_request.cache_paused_collection_details(
+            step=CurrentStep.post_webhooks,
+            collection=None,
+            action_needed=None,
+        )
+        privacy_request.status = PrivacyRequestStatus.paused
+        privacy_request.save(db=db)
+
+        queue_privacy_request(
+            privacy_request_id=privacy_request.id,
+            from_step=CurrentStep.post_webhooks.value,
+        )
+
+
+def initiate_scheduled_batch_consent_email_send() -> None:
+    """Initiates scheduler to add weekly batch consent email send"""
+
+    if CONFIG.is_test_mode:
+        return
+
+    logger.info("Initiating scheduler for batch consent email send")
+    scheduler.add_job(
+        func=send_consent_email_batch,
+        kwargs={},
+        id=BATCH_CONSENT_EMAIL_SEND,
+        coalesce=False,
+        replace_existing=True,
+        trigger="cron",
+        minute="0",
+        hour="12",
+        day_of_week="mon",
+        timezone="US/Eastern",
+    )
```

Comment:
This task is here for circular import reasons although if we rearranged the other ap scheduler task I think that could fix and make these tasks more visible.